### ISSUE TITLE
0008080:
fix ActiveSync redirection issues (on some servers)

### ISSUE THREAD

**Issue by Gloirin** [tine20/tine20#4079]
**Reported by konus on 20 Mar 2013 10:57**

**Version:** Kristina (2013.03.1)

1. the rewrite rule in the Wiki seems no to be up-to-date anymore
http://www.tine20.org/wiki/index.php/Admins/Synchronisation#Method_2:_Rewrite_configuration_by_a_.htaccess_file

2. (on some servers?!?) the rewrite rule is not enough to get to the &quot;it works&quot; display. It is necessary to change the code
\Tinebase\Core.php replace [&#039;REDIRECT_ACTIVESYNC&#039;] with [&#039;REDIRECT_REDIRECT_ACTIVESYNC&#039;] (2x) and
\ActiveSync\Server\Http.php  [&quot;REDIRECT_REMOTE_USER&quot;] with [&quot;REDIRECT_REDIRECT_REMOTE_USER&quot;] (2x)

**Steps to reproduce:** on my server

**Additional information:** see german thread http://www.tine20.org/forum/viewtopic.php?f=12&amp;t=10368

**Comment by Gloirin** [comment]
Related to #4415

**Comment by Gloirin** [comment]
**Comment posted by lkneschke on 20 Mar 2013 11:32**

I updated the rewrite rule in the wiki.

Can you please post your current rewrite rules.

Thanks.

**Comment by Gloirin** [comment]
**Comment posted by konus on 20 Mar 2013 12:22**

I addition to the above code changes I use the rules from /doc/htaccess without modification now:

RewriteEngine on
# ActiveSync
RewriteRule ^Microsoft-Server-ActiveSync index.php?frontend=activesync         [E=REMOTE_USER:%{HTTP:Authorization},L,QSA]

I is installed in a subdirectory html/tine/ on a shared host provider (www.sysprovide.de) with php 5.3.21 and Zend Engine v2.3.0, Suhosin v0.9.33

**Comment by Gloirin** [comment]
**Comment posted by lkneschke on 20 Mar 2013 12:41**

Can you please attach the modified version of Tinebase/Core.php too?

Thanks!

**Comment by Gloirin** [comment]
**Comment posted by konus on 20 Mar 2013 13:06**

Done :-)

**Comment by Gloirin** [comment]
**Comment posted by lkneschke on 20 Mar 2013 13:11**

If you change this lines

} elseif((isset($_SERVER[&#039;REDIRECT_REDIRECT_ACTIVESYNC&#039;]) &amp;&amp; $_SERVER[&#039;REDIRECT_REDIRECT_ACTIVESYNC&#039;] == &#039;true&#039;) ||
                 (isset($_GET[&#039;frontend&#039;]) &amp;&amp; $_GET[&#039;frontend&#039;] == &#039;activesync&#039;)) {

to

} elseif((isset($_GET[&#039;frontend&#039;]) &amp;&amp; $_GET[&#039;frontend&#039;] == &#039;activesync&#039;)) {

it does not work anymore?

I&#039;m curious because the rewrite rule does not set Apache variables anymore (S_SERVER*). Only the $_GET statement should match. Also your modifications should make no difference.

**Comment by Gloirin** [comment]
**Comment posted by konus on 2 Apr 2013 15:00**

Sorry for the delay:
Yes, I can confim, that changing tinebase/core.php to the above lines did help. I get a &quot;It works!&quot;.
I also get a &quot;it works!&quot; with the unchanged original, so I think changing this lines has NO EFFECT at all.

BUT ActiveSync\Server\Http.php has to be modified like in the attached file.

Sorry, I did not test it in this order. Since I did not understand what I did, I just followed the linked thread from the forum. 

Greetings
